clients/lsp-copilot.el

```diff
@@ -0,0 +1,250 @@
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;; Commentary:
+
+;; LSP client for the copilot node server -- https://www.npmjs.com/package/copilot-node-server
+
+;; Package-Requires: (lsp-mode secrets s compile dash cl-lib request company)
+
+;; Code:
+
+(require 'dash)
+(require 'lsp-mode)
+(require 's)
+
+(defgroup lsp-copilot ()
+  "Copilot LSP configuration"
+  :group 'lsp-mode
+  :tag "Copilot LSP"
+  :link '(url-link "https://www.npmjs.com/package/copilot-node-server"))
+
+(defcustom lsp-copilot-enabled t
+  "Whether the server should be started to provide completions."
+  :type 'boolean
+  :group 'lsp-copilot)
+
+(defcustom lsp-copilot-langserver-command-args '("--stdio")
+  "Command to start copilot-langserver."
+  :type '(repeat string)
+  :group 'lsp-copilot)
+
+(defcustom lsp-copilot-executable "copilot-lsp"
+  "The system-wise executable of lsp-copilot.
+When this executable is not found, you can stil use
+lsp-install-server to fetch an emacs-local version of the LSP."
+  :type 'string
+  :group 'lsp-copilot)
+
+
+(defcustom lsp-copilot-major-modes '(python-mode
+                                     python-ts-mode
+                                     go-mode
+                                     go-ts-mode
+                                     js-mode
+                                     js-ts-mode
+                                     java-mode
+                                     java-ts-mode
+                                     kotlin-mode
+                                     kotlin-ts-mode
+                                     ruby-mode
+                                     ruby-ts-mode
+                                     rust-mode
+                                     rust-ts-mode
+                                     tsx-ts-mode
+                                     typescript-mode
+                                     typescript-ts-mode
+                                     vue-mode
+                                     yaml-mode
+                                     yaml-ts-mode)
+
+  "The major modes for which lsp-copilot should be used"
+  :type '(repeat symbol)
+  :group 'lsp-copilot)
+
+(defcustom lsp-copilot-server-disabled-languages nil
+  "The lanuages for which the server must not be enabled (initialization setup for copilot)"
+  :type '(repeat string)
+  :group 'lsp-copilot)
+
+(defcustom lsp-copilot-server-multi-root t
+  "Whether the copilot server is started with multi-root"
+  :type 'boolean
+  :group 'lsp-copilot)
+
+(lsp-interface
```

Comment:
It looks like you will need to put this to `lsp-protocol`.
There're several ls already put their own interfaces there. Checkout `omnisharp:ErrorMessage` for example

Reply:
Done!